### PR TITLE
* NEW: Add XML configuration option `IncludeWixBinaries` to include Wix binaries in the analysis.

### PR DESCRIPTION
**Pull request by shaopeng-gh**
Added test file for Wix v4.
Test will only pass after my fix is merged: https://github.com/microsoft/binskim/pull/941

### CODE REVIEW

File: src/BinSkim.Sdk/BinaryAnalyzerContext.cs [hunk 1]
```diff
@@ -79,6 +79,8 @@ public CompilerDataLogger CompilerDataLogger
 
         public bool IgnorePdbLoadError { get; set; }
 
+        public bool IncludeWixBinaries { get; set; }
```
Comment (michaelcfanning) [inline src/BinSkim.Sdk/BinaryAnalyzerContext.cs:82]: >IncludeWixBinaries [](http://example.com/codeflow?start=20&length=18)

You didn't add a PerLanguageOption for this, as a result no one can configure this property via XML configuration.

**Comment by michaelcfanning** [comment]
>         public bool ComprehensiveBinaryParsing

Here is a correct example of property utilization.

---
Refers to: src/BinSkim.Sdk/BinaryAnalyzerContext.cs:43 in 60f6f7c. [](commit_id = 60f6f7c07402d13eabcd281b7d985c80804723ea, deletion_comment = False)

**Review by michaelcfanning — CHANGES_REQUESTED** [review]
:clock1:

**Comment by michaelcfanning** [comment]
>         public bool TracePdbLoads { get; set; }

add XML property

---
Refers to: src/BinSkim.Sdk/BinaryAnalyzerContext.cs:49 in 60f6f7c. [](commit_id = 60f6f7c07402d13eabcd281b7d985c80804723ea, deletion_comment = False)

**Comment by michaelcfanning** [comment]
>         public string SymbolPath { get; set; }

Add XML property

---
Refers to: src/BinSkim.Sdk/BinaryAnalyzerContext.cs:51 in 60f6f7c. [](commit_id = 60f6f7c07402d13eabcd281b7d985c80804723ea, deletion_comment = False)

**Comment by michaelcfanning** [comment]
>                     ? this.Policy.GetProperty(SharedCompilerDataLoggerProperty)

remove xml property

---
Refers to: src/BinSkim.Sdk/BinaryAnalyzerContext.cs:74 in 60f6f7c. [](commit_id = 60f6f7c07402d13eabcd281b7d985c80804723ea, deletion_comment = False)

File: src/BinSkim.Sdk/BinaryAnalyzerContext.cs (hunk 1 above)
Comment (shaopeng-gh) [reply src/BinSkim.Sdk/BinaryAnalyzerContext.cs:82]: fixed

File: src/BinSkim.Driver/AnalyzeOptions.cs [hunk 2]
```diff
@@ -52,5 +52,10 @@ public new IEnumerable<string> Trace
             "ignorePdbLoadError",
             HelpText = "If enabled, BinSkim won't break if we have a 'PdbLoadingException'.")]
         public bool IgnorePdbLoadError { get; set; }
+
+        [Option(
+            "includeWixBinaries",
+            HelpText = "If enabled, BinSkim will include Wix binaries in the analysis.")]
+        public bool IncludeWixBinaries { get; set; }
```
[the remark this answers is not in the document]
Comment (shaopeng-gh) [reply src/BinSkim.Driver/AnalyzeOptions.cs:59]: as last conversation in the sync meeting, remove the option and use the xml config

**Comment by shaopeng-gh** [comment]
>         public bool TracePdbLoads { get; set; }

added

---
In reply to: [1640531872](https://github.com/microsoft/binskim/pull/944#issuecomment-1640531872) [](http://example.com/codeflow?ancestors=1640531872,1640531872)

---
Refers to: src/BinSkim.Sdk/BinaryAnalyzerContext.cs:49 in 60f6f7c. [](commit_id = 60f6f7c07402d13eabcd281b7d985c80804723ea, deletion_comment = False)

**Comment by shaopeng-gh** [comment]
>         public bool ComprehensiveBinaryParsing

Using this format now, thanks!

---
In reply to: [1640480894](https://github.com/microsoft/binskim/pull/944#issuecomment-1640480894) [](http://example.com/codeflow?ancestors=1640480894,1640480894)

---
Refers to: src/BinSkim.Sdk/BinaryAnalyzerContext.cs:43 in 60f6f7c. [](commit_id = 60f6f7c07402d13eabcd281b7d985c80804723ea, deletion_comment = False)

File: src/BinSkim.Sdk/BinaryAnalyzerContext.cs [hunk 3]
```diff
@@ -79,6 +79,8 @@ public CompilerDataLogger CompilerDataLogger
 
         public bool IgnorePdbLoadError { get; set; }
```
[the remark this answers is not in the document]
Comment (shaopeng-gh) [reply src/BinSkim.Sdk/BinaryAnalyzerContext.cs:80]: added

**Comment by shaopeng-gh** [comment]
>         public string LocalSymbolDirectories { get; set; }

added

---
In reply to: [1640531469](https://github.com/microsoft/binskim/pull/944#issuecomment-1640531469) [](http://example.com/codeflow?ancestors=1640531469,1640531469)

---
Refers to: src/BinSkim.Sdk/BinaryAnalyzerContext.cs:41 in 60f6f7c. [](commit_id = 60f6f7c07402d13eabcd281b7d985c80804723ea, deletion_comment = False)

**Comment by shaopeng-gh** [comment]
>         public string SymbolPath { get; set; }

added

---
In reply to: [1640532133](https://github.com/microsoft/binskim/pull/944#issuecomment-1640532133) [](http://example.com/codeflow?ancestors=1640532133,1640532133)

---
Refers to: src/BinSkim.Sdk/BinaryAnalyzerContext.cs:51 in 60f6f7c. [](commit_id = 60f6f7c07402d13eabcd281b7d985c80804723ea, deletion_comment = False)

**Comment by shaopeng-gh** [comment]
>         public CompilerDataLogger CompilerDataLogger

removed

---
In reply to: [1640480352](https://github.com/microsoft/binskim/pull/944#issuecomment-1640480352) [](http://example.com/codeflow?ancestors=1640480352,1640480352)

---
Refers to: src/BinSkim.Sdk/BinaryAnalyzerContext.cs:69 in 60f6f7c. [](commit_id = 60f6f7c07402d13eabcd281b7d985c80804723ea, deletion_comment = False)

**Comment by shaopeng-gh** [comment]
>                     ? this.Policy.GetProperty(SharedCompilerDataLoggerProperty)

removed

---
In reply to: [1640532812](https://github.com/microsoft/binskim/pull/944#issuecomment-1640532812) [](http://example.com/codeflow?ancestors=1640532812,1640532812)

---
Refers to: src/BinSkim.Sdk/BinaryAnalyzerContext.cs:74 in 60f6f7c. [](commit_id = 60f6f7c07402d13eabcd281b7d985c80804723ea, deletion_comment = False)

**Comment by shaopeng-gh** [comment]
>         public bool TracePdbLoads { get; set; }

fixed all others, this one is a bit different, it depends on the --trace, proposed to leave it another pr.

---
In reply to: [1640531872](https://github.com/microsoft/binskim/pull/944#issuecomment-1640531872) [](http://example.com/codeflow?ancestors=1640531872,1640531872)

---
Refers to: src/BinSkim.Sdk/BinaryAnalyzerContext.cs:49 in 60f6f7c. [](commit_id = 60f6f7c07402d13eabcd281b7d985c80804723ea, deletion_comment = False)

**Review by michaelcfanning — APPROVED** [review]
:shipit:

**Pull request merged**